### PR TITLE
Add lineSpacing prop to Android.

### PR DESCRIPTION
**Pull request by zezhouliu**
## Summary
This PR makes enables `lineSpacing` on Android.  It is based on https://github.com/facebook/react-native/pull/9420, differing only with respect to minor library import references.

### Android fonts on Airbnb app
screen shot 2017-01-23 at 9 23 25 pm

### Android fonts specific use cases:

screen shot 2017-01-23 at 9 25 30 pm

screen shot 2017-01-23 at 9 27 42 pm

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java [hunk 1]
```diff
@@ -64,6 +67,18 @@ public void setText(ReactTextUpdate update) {
...
 
+    float nextLetterSpacing = update.getLetterSpacing();
+    int fontSize = update.getFontSize();
+    if (!FloatUtil.floatsEqual(mLetterSpacing, nextLetterSpacing)) {
+      mLetterSpacing = nextLetterSpacing;
+      if(Float.isNaN(mLetterSpacing)) {
+        setLetterSpacing((float)0.0);
+      } else {
+        //calculate EM from proper font pixels
+        setLetterSpacing(1+(mLetterSpacing-PixelUtil.toDIPFromPixel(fontSize))/PixelUtil.toDIPFromPixel(fontSize));
```
Comment (lelandrichardson) [inline ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java]: I think `setLetterSpacing` isn't available on all android versions. Can you run this in Android 4.x and see if everything builds? We don't need the letter spacing to work in that case, but it shouldn't break anything either.

**Comment by spikebrehm** [comment]
@zezhouliu thanks for the PR!  It scares me a bit to allow our fork to continue to drift from origin.  Are you planning on reviving the original PR to `facebook/react-native`? That would make me more comfortable and would make maintenance of the fork easier going forward.

**Comment by spikebrehm** [comment]
BTW after you merge this, I can help you go through the rigamarole of creating a new patch version and using this in our codebase.  I have a doc I'll send you with the steps.

**Comment by zezhouliu** [comment]
@spikebrehm : yes, i'm planning on reviving the original PR to merge into fb!

**Comment by lelandrichardson** [comment]
@zezhouliu can you address my question about letter spacing?  If it requires an android API of a certain version, I'm hoping we can fence the callsites with an SDK version check.

**Comment by lelandrichardson** [comment]
@zezhouliu any updates?

**Comment by zezhouliu** [comment]
https://github.com/airbnb/react-native/pull/13

**Pull request closed without merging**